client/dashboard/app/queries/me-preferences.ts

```diff
@@ -2,30 +2,41 @@ import { fetchPreferences, updatePreferences } from '../../data/me-preferences';
 import { queryClient } from '../query-client';
 import type { UserPreferences } from '../../data/me-preferences';
 
-export const userPreferencesQuery = (
-	preferenceName?: keyof UserPreferences,
-	defaultValue?: UserPreferences[ keyof UserPreferences ]
+const defaultValues: UserPreferences = {
+	'sites-view': {},
+	'some-string': '',
+};
+
+type QueryGetSetData< P > = P extends keyof UserPreferences
```

Comment:
This is getting more complex and the benefit does not really outweigh the poor legibility of the typings 😅 I agree with your suggestion of splitting the queries.

Reply:
_And_ we can remove the `as any` casts this way too!

Reply:
I've split the `userPreferencesQuery` in two.